### PR TITLE
Java Input Output Exercises

### PR DESCRIPTION
**Pull request by dskrumov**
@martinmilev make a review.

### CODE REVIEW

**Comment by mgenov** [comment]
Commits are still not squashed.

File: src/com/company/streams/ObjectTransfer.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    private InputStream inputStream;
+    private OutputStream outputStream;
+
+    /**
+     * Read from the array which is defined in main method and writes the data with specific length
+     * and offset in the streamTEST.txt
+     *
+     * @throws IOException
+     */
+    public void transfer(String inputFileName, int offset, int length) throws IOException {
```
Comment (mgenov) [inline src/com/company/streams/ObjectTransfer.java]: This is what task states about the signature of the method: 
```java
class TransferObject {
   public int transfer(InputStream in, OutputStream out, int offset, int size) {
      // 
   }
}
```

As you can see the example signature returns the number of bytes that are transferred which is use-full for the client code which uses this method.

I'm not sure that your implementation will work as expected as it writes at offset and is not reading. 

If I have a file with bytes:
01234567890

When I want to transfer from offset: 3 and limit of 5 bytes I should get result like:
34567 in the output file.

File: src/com/company/streams/ObjectTransfer.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class ObjectTransfer {
+    /**
+     * Fields
+     */
+    private InputStream inputStream;
```
Comment (mgenov) [inline src/com/company/streams/ObjectTransfer.java]: You don't need these references as they are used on local level in the method.

File: src/com/company/reversedocument/File.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    private FileReader fileReader;
+    private BufferedReader bufferedReader;
+
+    /**
+     * Create the file
+     *
+     * @param documentReadName is the name of the created file
+     */
+    public File(String documentReadName) throws FileNotFoundException {
```
Comment (mgenov) [inline src/com/company/reversedocument/File.java]: inputFile instead of documentReadName

File: src/com/company/reversedocument/File.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
+package com.company.reversedocument;
+
+import java.io.*;
+
+public class File {
+    /**
+     * Fields
+     */
+
+    private FileReader fileReader;
```
Comment (mgenov) [inline src/com/company/reversedocument/File.java]: You can keep just name of the file as local variable and use locals for BufferedReader in the reverseInfo/reverseContent method.

File: src/com/company/reversedocument/File.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+
+    /**
+     * Reverse the contend from the previous created file by reading it
+     *
+     * @param documentWriteName is the name of the second document which will store the reversed information
+     * @throws IOException
+     */
+    public void reverseInto(String documentWriteName) throws IOException {
```
Comment (mgenov) [inline src/com/company/reversedocument/File.java]: reverseContentInto(String outputFile) sounds better.

**Comment by dskrumov** [comment]
Its still not ready for review. I made the last exercise and pushed to the repository because i want to clone it at home and work on it.

**Pull request closed without merging**